File: app/src/main/java/com/quiz/mallzellij_show_android/model/ArticleStock.java
package com.quiz.mallzellij_show_android.model;

import java.math.BigDecimal;

public class ArticleStock {
    private Long siteId;
    private String siteName;
    private Integer quantity;
    private Integer quantiteALouer;
    private Integer prix;
    private BigDecimal prixPromo;
    private BigDecimal prixPrevendor;

    public Long getSiteId() { return siteId; }
    public String getSiteName() { return siteName; }
    public Integer getQuantity() { return quantity; }
    public Integer getQuantiteALouer() { return quantiteALouer; }
    public Integer getPrix() { return prix; }
    public BigDecimal getPrixPromo() { return prixPromo; }
    public BigDecimal getPrixPrevendor() { return prixPrevendor; }
}

File: app_flutter/lib/models/article_stock.dart
class ArticleStock {
  final int? siteId;
  final String? siteName;
  final int? quantity;
  final int? quantiteALouer;
  final double? prix;

  ArticleStock({this.siteId, this.siteName, this.quantity, this.quantiteALouer, this.prix});

  factory ArticleStock.fromJson(Map<String, dynamic> json) => ArticleStock(
    siteId: json['siteId'] as int?,
    siteName: json['siteName'] as String?,
    quantity: json['quantity'] as int?,
    quantiteALouer: json['quantiteALouer'] as int?,
    prix: (json['prix'] as num?)?.toDouble(),
  );
}
